portal-impl/src/com/liferay/portal/service/impl/PermissionPropagationEntryLocalServiceImpl.java

```diff
@@ -34,6 +55,22 @@ public PermissionPropagationEntry fetchPermissionPropagationEntry(
 			_classNameLocalService.getClassNameId(className), classPK);
 	}
 
+	public void updatePermissionPropagationEntry(
+		long companyId, long groupId, String className, long classPK,
+		boolean propagation) {
+
+		PermissionPropagationEntry permissionPropagationEntry =
+			fetchPermissionPropagationEntry(
+				companyId, groupId, className, classPK);
+
+		if (permissionPropagationEntry != null) {
+			permissionPropagationEntry.setPropagation(propagation);
+
+			permissionPropagationEntryPersistence.update(
+				permissionPropagationEntry);
+		}
+	}
+
```

Comment:
Does it work for the Home folder and pre-existing folder? Look like with this we can't update propagation for them. Please help double-check. Thanks.

Reply:
Fixed it! Thanks